### PR TITLE
Add filter positional argument to clusters list

### PR DESCRIPTION
**Pull request by cblecker**
If there is a positional argument, use it as a name or ID filter. Compatible with the `--managed` flag too.

### CODE REVIEW

File: cmd/uhc/cluster/list/list.go [hunk 1]
```diff
@@ -124,6 +120,12 @@ func run(cmd *cobra.Command, argv []string) {
 		managed = false
 	}
 
+	// if there is a parameter specified, assume its a filter
```
Comment (jhernand) [inline cmd/uhc/cluster/list/list.go]: I try to stick to comments that start with upper case and end with punctuation: `If there ... a filter:`. Can you do the same?

File: cmd/uhc/cluster/list/list.go [hunk 2]
```diff
@@ -46,10 +46,11 @@ var managed bool
 
 // Cmd Constant:
 var Cmd = &cobra.Command{
-	Use:   "list [flags] ",
+	Use:   "list [flags] [partial cluster ID or name]",
 	Short: "List clusters",
 	Long:  "List clusters by ID and Name",
-	Run:   run,
+	Args:  cobra.RangeArgs(0, 1),
+	RunE:  run,
```
Comment (jhernand) [inline cmd/uhc/cluster/list/list.go:53]: This change to `RunE` is nice. Is there any chance that you can do the same in all the other commands?

**Review by jhernand — COMMENTED** [review]
Looks good. Just a very minor suggestion about comments.

@ALimaRedHat please take a look.

File: cmd/uhc/cluster/list/list.go (hunk 1 above)
Comment (cblecker) [reply cmd/uhc/cluster/list/list.go]: Comment fixed

File: cmd/uhc/cluster/list/list.go (hunk 2 above)
Comment (cblecker) [reply cmd/uhc/cluster/list/list.go:53]: Absolutely. Would that be better done in a follow up PR though (as to not balloon the size of this PR)?

Comment (jhernand) [reply cmd/uhc/cluster/list/list.go:53]: Yes, sure. A separate PR is better.

**Review by jhernand — APPROVED** [review]
Looks good, thanks @cblecker. I will wait a bit for @ALimaRedHat to review.

**Review by ghost — APPROVED** [review]
Looks good to me too

**Review by fahlmant — APPROVED** [review]

**Pull request merged**